### PR TITLE
Document issue & PR templates

### LINKED ISSUE #86: Add section on PR and issue templates to maintenance docs
Cf. the maintenance [README in the docs folder (TODOs as comments)](https://raw.githubusercontent.com/hexatomic/hexatomic/develop/docs/dev/src/maintenance/README.md)

### PR DESCRIPTION
**Pull request by sdruskat**
## Pull request type

 

- Documentation content changes

## What is the current behavior?

Issue Number: #86

The use of PR and issue templates are undocumented.

## What is the new behavior?

- Start a new section in the maintenance section of the dev docs for
repository documentation
- Document issue and PR templates
- Fixes #86

## Does this introduce a breaking change?

No

## Other information

N/A

## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] Tests for the changes have been added **or** the PR is neither a bug fix nor a feature
- [x] Docs have been reviewed and added / updated if needed **or** the PR is neither a bug fix nor a feature
- [x] The *Unreleased* section in CHANGELOG.md has been amended to reflect the changes in this PR if needed
- [x] Build (`mvn verify`) was run locally and any changes were pushed
- [x] The pull request is against the correct branch (`master` for bug fixes, `develop` for new functionality)
- [x] [Dependencies and citation templates](https://github.com/hexatomic/hexatomic/tree/develop/releng/templates) have been updated where necessary **or** there are no new dependencies

---

# Checklist for maintainers

## Releases

- Do **NOT** push the green merge button on GitHub.  
Follow the latest version of the [developer/maintainer docs](https://hexatomic.github.io/hexatomic/dev/) for making releases.
- Check that license and citation information for dependencies are complete (each dependency has a folder in `/THIRD-PARTY/`).

### CODE REVIEW

File: docs/dev/src/maintenance/repository/README.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# Repository maintenance
+
+The Hexatomic source code is hosted on GitHub at <i class="fa fa-github"></i> <https://github.com/hexatomic/hexatomic> (the *repository*).
+This section documents the maintanence of the repository.
```
Comment (thomaskrause) [inline docs/dev/src/maintenance/repository/README.md:4]:
```suggestion
This section documents the maintenance of the repository.
```

Typo

File: docs/dev/src/maintenance/repository/templates.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+Further instructions for users are included as `<!-- HTML comments -->`.
+
+## Issue templates
+
+You can find the issue templates in `.github/ISSUE_TEMPLATE/`.
+GitHub will use the contents of this directory to determine the way that new issues are created.
+
+The Hexatomic repository has two issue templates, one for bug reports ([`.github/ISSUE_TEMPLATE/bug_report.md`]()), and one for feature requests ([`.github/ISSUE_TEMPLATE/feature_request.md`]()).
+Therefore, when you create a new issue, you are being asked which type of issue you want to create: a bug report, or a feature request.
+The names and descriptions of the options come from the YAML frontmatter of the different issue template files.
```
Comment (thomaskrause) [inline docs/dev/src/maintenance/repository/templates.md:24]:
```suggestion
The names and descriptions of the options come from the YAML front matter of the different issue template files.
```

File: docs/dev/src/maintenance/repository/templates.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
+# Template files
+
+Template files can be used to customize the GitHub repository to some extent.
+The Hexatomic repository uses two different template types:
+
+1. Issue templates
+2. A pull request template
+
+These templates are text files in the `.github/` directory that are formatted with Markdown, and may have a [YAML](https://yaml.org/) frontmatter.
```
Comment (thomaskrause) [inline docs/dev/src/maintenance/repository/templates.md:9]:
```suggestion
These templates are text files in the `.github/` directory that are formatted with Markdown, and may have a [YAML](https://yaml.org/) front matter.
```
At least that's what [Wikipedia](https://en.wikipedia.org/wiki/Book_design#Front_matter), [Merriam-Webster](https://www.merriam-webster.com/dictionary/front%20matter) and several English (GB) - German dictionaries are using (Cambridge Dictionary somehow does not have this entry and I won't buy a subscription to the Oxford Dictionary just to win an argument :wink:  )

File: docs/dev/src/maintenance/repository/templates.md [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+Therefore, when you create a new issue, you are being asked which type of issue you want to create: a bug report, or a feature request.
+The names and descriptions of the options come from the YAML frontmatter of the different issue template files.
+
+![The user is presented with a choice of issue templates when creating a new issue.](template-choice.png)
+
+On a side note, if there is only one template per template type (issue/pull request), no name and description is needed, and the single file will automatically be used.
+
+Once you choose between issue types, a new issue will be created for you to edit.
+The editor will be pre-populated with the contents of the respective Markdown file.
+Additionally, the frontmatter of the respective Markdown file determines some other values for the newly created issue, for example
```
Comment (thomaskrause) [inline docs/dev/src/maintenance/repository/templates.md:32]:
```suggestion
Additionally, the front matter of the respective Markdown file determines some other values for the newly created issue, for example
```

File: docs/dev/src/maintenance/repository/templates.md [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+
+- the name of the template (shown in the issue choice dialogue);
+- the description of the template (shown in the issue choice dialogue)
+- the title of the newly created issue
+- A list of labels that will be assigned to the newly creted issue
+- A list of assignees that the newly created issue will be assigned to.
+
+Below is an example for the frontmatter of the bug report template. 
+The triple hyphens define a section of frontmatter in YAML.
```
Comment (thomaskrause) [inline docs/dev/src/maintenance/repository/templates.md:41]:
```suggestion
The triple hyphens define a section of front matter in YAML.
```

File: docs/dev/src/maintenance/repository/templates.md [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+---
+name: Bug report
+about: Create a report to help us improve Hexatomic.
+title: ''
+labels: 'bug'
+assignees: ''
+---
+```
+
+This frontmatter, together with the Markdown body text of the template creates a pre-populated new issue as in the following figure.
```
Comment (thomaskrause) [inline docs/dev/src/maintenance/repository/templates.md:53]:
```suggestion
This front matter, together with the Markdown body text of the template creates a pre-populated new issue as in the following figure.
```

File: docs/dev/src/maintenance/repository/templates.md [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+
+![This frontmatter, together with the Markdown body text of the template creates a pre-populated new issue.](new-bug-report.png)
```
Comment (thomaskrause) [inline docs/dev/src/maintenance/repository/templates.md:55]:
```suggestion
![This front matter, together with the Markdown body text of the template creates a pre-populated new issue.](new-bug-report.png)
```

File: docs/dev/src/maintenance/repository/templates.md [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+The editor will be pre-populated with the contents of the respective Markdown file.
+Additionally, the frontmatter of the respective Markdown file determines some other values for the newly created issue, for example
+
+- the name of the template (shown in the issue choice dialogue);
+- the description of the template (shown in the issue choice dialogue)
+- the title of the newly created issue
+- A list of labels that will be assigned to the newly creted issue
+- A list of assignees that the newly created issue will be assigned to.
+
+Below is an example for the frontmatter of the bug report template.
```
Comment (thomaskrause) [inline docs/dev/src/maintenance/repository/templates.md:40]:
```suggestion
Below is an example for the front matter of the bug report template. 
```

File: docs/dev/src/maintenance/repository/templates.md [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+The Hexatomic repository uses two different template types:
+
+1. Issue templates
+2. A pull request template
+
+These templates are text files in the `.github/` directory that are formatted with Markdown, and may have a [YAML](https://yaml.org/) frontmatter.
+You can learn more about issue templates and pull request templates from the [GitHub documentation](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/about-issue-and-pull-request-templates).
+
+The templates contain the predefined contents of the initial issue comment, or the initial pull request comment.
+They may contain checklists, text sections, bullet lists, etc., that work as "form elements" the user is aksed to complete.
```
Comment (thomaskrause) [inline docs/dev/src/maintenance/repository/templates.md:13]: Typo
```suggestion
They may contain checklists, text sections, bullet lists, etc., that work as "form elements" the user is asked to complete.
```

File: docs/dev/src/maintenance/repository/templates.md [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+On a side note, if there is only one template per template type (issue/pull request), no name and description is needed, and the single file will automatically be used.
+
+Once you choose between issue types, a new issue will be created for you to edit.
+The editor will be pre-populated with the contents of the respective Markdown file.
+Additionally, the frontmatter of the respective Markdown file determines some other values for the newly created issue, for example
+
+- the name of the template (shown in the issue choice dialogue);
+- the description of the template (shown in the issue choice dialogue)
+- the title of the newly created issue
+- A list of labels that will be assigned to the newly creted issue
```
Comment (thomaskrause) [inline docs/dev/src/maintenance/repository/templates.md:37]: Typo
```suggestion
- A list of labels that will be assigned to the newly created issue
```

**Review by thomaskrause — APPROVED** [review]
Good to go, but I found some typos

**Comment by sdruskat** [comment]
> 
> 
> Good to go, but I found some typos

Thanks fixed. Please merge if checks pass.

**Pull request merged**